code/modules/mob/say.dm

```diff
@@ -78,6 +78,8 @@
 	var/source = "<span class='game'><span class='prefix'>DEAD:</span> <span class='name'>[name]</span>[alt_name]"
 	var/rendered = " <span class='message'>[emoji_parse(spanned)]</span></span>"
 	log_talk(message, LOG_SAY, tag="DEAD")
+	if(SEND_SIGNAL(src, COMSIG_MOB_DEADSAY, src, message) | MOB_DEADSAY_SIGNAL_INTERCEPT)

```

Comment:
src as the first argument is not necessary, it gets added on its own

Also this will always be true, you want `&` to check if the return flag is sent
```suggestion
	if(SEND_SIGNAL(src, COMSIG_MOB_DEADSAY, message) & MOB_DEADSAY_SIGNAL_INTERCEPT)
```